x-pack/plugins/security_solution/public/resolver/store/middleware/index.ts

```diff
@@ -5,21 +5,52 @@
  * 2.0.
  */
 
-import type { Dispatch, MiddlewareAPI } from 'redux';
-import type { ResolverState, DataAccessLayer } from '../../types';
+import type { Dispatch, MiddlewareAPI, AnyAction } from 'redux';
+import { isEmpty } from 'lodash';
+import type { DataAccessLayer, AnalyzerById } from '../../types';
 import { ResolverTreeFetcher } from './resolver_tree_fetcher';
-
-import type { ResolverAction } from '../actions';
+import type { State } from '../../../common/store/types';
 import { RelatedEventsFetcher } from './related_events_fetcher';
 import { CurrentRelatedEventFetcher } from './current_related_event_fetcher';
 import { NodeDataFetcher } from './node_data_fetcher';
+import * as Actions from '../actions';
+import * as DataActions from '../data/action';
+import * as CameraActions from '../camera/action';
 
-type MiddlewareFactory<S = ResolverState> = (
+type MiddlewareFactory<S = State> = (
   dataAccessLayer: DataAccessLayer
 ) => (
-  api: MiddlewareAPI<Dispatch<ResolverAction>, S>
-) => (next: Dispatch<ResolverAction>) => (action: ResolverAction) => unknown;
+  api: MiddlewareAPI<Dispatch<AnyAction>, S>
+) => (next: Dispatch<AnyAction>) => (action: AnyAction) => unknown;
+
+const resolverActions = { ...Actions, ...DataActions, ...CameraActions };
+
+/**
+ * Helper function to determine if analyzer is active (resolver middleware should be run)
+ * analyzer is considered active if:
+ * 1. action is initial set up (create resolver), or
+ * 2. action is not clean up, or
+ * 3. analyzer state for action id is not empty
+ * @param state analyzerbyId state
+ * @param action dispatched action
+ * @returns boolean of whether the analyzer of id has an store in redux
+ */
+function isAnalyzerActive(state: AnalyzerById, action: AnyAction): boolean {
+  // middleware shouldn't run after clear resolver
+  if (Actions.clearResolver.match(action)) {
+    return false;
+  }
+  return Actions.createResolver.match(action) || !isEmpty(state[action.payload?.id]);
+}
 
+/**
+ * Helper function to check whether an action is a resolver action
+ * @param action dispatched action
+ * @returns boolean of whether the action is a resolver action
+ */
+function isResolverAction(action: AnyAction): boolean {
+  return Object.values(resolverActions).some((x) => x.match(action));
```

Comment:
Resolver can fire a lot of actions (maybe 1000s per second) if I recall correctly. This function might be worth optimizing or at least profiling.